spyder/utils/introspection/manager.py

```diff
@@ -154,7 +183,18 @@ def _handle_timeout(self):
         self.waiting = False
         if self.pending:
             self._finalize(self.pending)
+        elif self.pending_request:
+            info = self.pending_request
+            self.pending_request = None
+            self.send_request(info)
         else:
+            self.timeout_counter += 1
+            if (self.timeout_counter > MAX_TIMEOUTS and
+                    not self.plugins_initialized['jedi']):

```

Comment:
1. Then that `MAX_TIMEOUTS=15` is too much. It should be 5, or even 3. It was like that before, why do you change it?
2. That's odd, but we can take a look at that in the future.